stream_alert/apps/_apps/intercom.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+import requests
+
+from . import AppIntegration, StreamAlertApp, get_logger
+
+
+LOGGER = get_logger(__name__)
+
+
+@StreamAlertApp
+class IntercomApp(AppIntegration):
+    """Intercom StreamAlert app"""
+    _INTERCOM_LOGS_URL = 'https://api.intercom.io/admins/activity_logs'
+
+    def __init__(self, event, config):
+        super(IntercomApp, self).__init__(event, config)
+        self._next_page = None
+
+    @classmethod
+    def _type(cls):
+        return 'admin_activity_logs'
+
+    @classmethod
+    def service(cls):
+        return 'intercom'
+
+    @classmethod
+    def _required_auth_info(cls):
+        return {
+            'token': {
+                'description': 'the OAuth token for this Intercom app',
+                'format': re.compile(r'.*')
+            }
+        }
+
+    def _sleep_seconds(self):
+        """Return the number of seconds this polling function should sleep for
+        between requests to avoid failed requests. Intercom API allows for a default of 500 requests
+        every minute, distributed over 10 seconds periods. This means for a default rate limit of
+        500 per minute, you can send a maximum of 83 operations per 10 second period.
+
+        Resource(s):
+            https://developers.intercom.com/intercom-api-reference/reference#rate-limiting
+
+        Returns:
+            int: Number of seconds that this function should sleep for between requests
+        """
+        return 10
+
+    def _gather_logs(self):
+        # Generate headers
+        headers = {'Authorization': "Bearer %s" % self._config.auth['token'],
+                   'Accept': 'application/json'}
+
+        # Results are paginated with a page url field provided that is used in subsequent queries.
+        # If this field exists, make a a query to the page url, and if not, make a fresh query to
+        # the default _INTERCOM_LOGS_URL with the required created_at_before and created_at_after
+        # parameters
+        if self._next_page:
+            params = None
+            url = self._next_page
+        else:
+            params = {'created_at_before': int(time.time()),
+                      'created_at_after': self._last_timestamp}
+            url = self._INTERCOM_LOGS_URL
+
+        LOGGER.info("Requesting events from: %s params: %s", url, params)
+
+        try:
+            result, response = self._make_get_request(url, headers=headers, params=params)
+        except requests.exceptions.ConnectionError:
+            LOGGER.exception('Received bad response from Intercom')
+            return False
+
+        if not result:
+            return False
+
+        activities = [
+            activity
+            for activity
+            in response['activity_logs']
+            if int(activity['created_at']) > self._last_timestamp]
+
+        if not activities:
+            return False
+
+        # Save next page url if any for paginated results
+        if response['pages']['next'] is not None:
```

Comment:
is will the `next` key actually exist even when the value is `None` (ie - is this a non-optional key?)

If it's optional, you might want to do `response['pages'].get('next')` but I will defer to you since I do not know their API.

Reply:
`next` is a non-optional key -- it'll always exist and it's value can be `None`

Reply:
awesome thanks for confirming!